Codec/Archive/Zip/Internal.hs

```diff
@@ -185,9 +189,16 @@ zipVersion = Version [4,6] []
 -- library.
 
 scanArchive
+  :: Either Handle (Path Abs File)     -- ^ Path to archive to scan
+  -> IO (ArchiveDescription, Map EntrySelector EntryDescription)
+scanArchive (Right path) = withFile (toFilePath path) ReadMode $ __scanArchive path
+scanArchive (Left h) = __scanArchive blindPath h
+
+__scanArchive
```

Comment:
Please don't use `__xxx` naming convention because that is not how the rest of the code in named.